File: android/ValleTPV/ValleTPVlib/src/main/java/com/valleapp/valletpvlib/CashlogyManager/CashlogyAction.java
package com.valleapp.valletpvlib.CashlogyManager;

public abstract class CashlogyAction {
    protected CashlogySocketManager socketManager;

    public CashlogyAction(CashlogySocketManager socketManager) {
        this.socketManager = socketManager;
    }

    // Método para ejecutar la acción. Debe ser implementado por las clases que extienden CashlogyAction.
    public abstract void execute();

    // Método para manejar la respuesta recibida. Debe ser implementado por las clases que extienden CashlogyAction.
    public abstract void handleResponse(String command, String response);
}

File: android/ValleTPV/ValleTPVlib/src/main/java/com/valleapp/valletpvlib/cashlogymanager/CashlogyAction.kt
package com.valleapp.valletpvlib.cashlogymanager

import android.os.Bundle
import android.os.Handler
import android.os.Message

abstract class CashlogyAction(
    protected val socketManager: CashlogySocketManager,
    protected  val uiHandler: Handler) {
    abstract fun execute()
    // Función 'open' para notificar al UI handler
    open fun notifyUI(key: String, value: String) {
        val msg = Message().apply {
            data = Bundle().apply {
                putString("key", key)
                putString("value", value)
            }
        }
        uiHandler.sendMessage(msg)
    }
}

